CodeConverter/CSharp/MethodBodyExecutableStatementVisitor.cs

```diff
@@ -123,19 +123,30 @@ public override async Task<SyntaxList<StatementSyntax>> VisitLocalDeclarationSta
                     _perScopeState.HoistToTopLevel(new HoistedFieldFromVbStaticVariable(methodName, variable.Identifier.Text, initializeValue, decl.Declaration.Type, isVbShared));
                 }
             } else {
-                foreach (var decl in localDeclarationStatementSyntaxs) {
-                    if (_perScopeState.IsInsideLoop() && declarator.Initializer is null && declarator.AsClause is not VBSyntax.AsNewClauseSyntax) {
-                        foreach (var variable in decl.Declaration.Variables) {
-                            _perScopeState.Hoist(new HoistedDefaultInitializedLoopVariable(
-                                variable.Identifier.Text,
-                                // e.g. "b As Boolean" has no intializer but can turn into "var b = default(bool)"
-                                variable.Initializer?.Value,
-                                decl.Declaration.Type,
-                                _perScopeState.IsInsideNestedLoop()));
+                var shouldPullVariablesBeforeLoop = _perScopeState.IsInsideLoop() && declarator.Initializer is null && declarator.AsClause is not VBSyntax.AsNewClauseSyntax;
+                if (shouldPullVariablesBeforeLoop) {
+                    foreach (var variablesDecl in variables) {
+                        var newVariableDecl = variablesDecl.Decl;
+                        foreach (var (csVariable, vbVariable) in variablesDecl.Decl.Variables.Zip(variablesDecl.VbVariables, Tuple.Create)) {
```

Comment:
I don't think you can rely on zipping these back together since I think `SplitVariableDeclarationsAsync` can convert some into a a local function in `methods` instead of in `variables`

Reply:
I think it will be fine but I can add a test for it.

Reply:
Removed zip.
Added test when local variable in loop in VB becomes a function in C#.